### DISCUSSION TITLE
Control the order of routes

### DISCUSSION THREAD

**Discussion by mszkb** [posva/unplugin-vue-router#357] [Questions and Help] [answered]
Is there a way to control the order of the (nested) routes - besides changing the file/dir name?

For example I have this structure:

![image](https://github.com/posva/unplugin-vue-router/assets/10475327/97581cd0-191c-417b-8432-90647c44acde)

And then I want to generate a navigation component based on my routes by using:

````
const router = useRouter();
const routes = router.getRoutes();
````

This gives me following:

![image](https://github.com/posva/unplugin-vue-router/assets/10475327/8d0bc316-a5d7-431d-a8a9-564b06fc7f97)

My goal is to have logs at the end of the list:
1. /
2. /system-library
3. /sign-configuration
4. /logs

**Comment by posva** [comment, marked as answer]
You have to define a [custom meta property](https://uvr.esm.is/guide/extending-routes.html#definepage) that allows you to sort the routes. You could for example add an `order` property and then use it in a `getRoutes().sort()`

**Reply by mszkb** [reply]
Yeah I thought so, that this is kinda out of scope of this plugin.

I ended up doing:

````
<route lang="yaml">
meta:
  order: 1
  title: Sign Monitoring
</route>
````

and then handle the sorting after meta.order in my composable.
